pyat/at/lattice/lattice_object.py

```diff
@@ -328,6 +328,31 @@ def extend(self, elems: Iterable[Element]):
     def append(self, elem: Element):
         self.extend([elem])
 
+    def develop(self, periods: Optional[int] = None) -> "Lattice":
+        """Develop a periodical lattice by repeating its elements
+        *self.periodicity* times
+
+        The elements of the new lattice are deep copies ot the original
+        elements, so that they are all independent.
+
+        Args:
+            periods: The lattice is repeated *periods* times instead of
+              *self.periodicity*
+
+        Returns:
+            newlattice: The developed lattice
+        """
+        if periods is None:
+            periods = self.periodicity
+        newper, rem = divmod(self.periodicity, periods)
```

Comment:
> Also this function will never work if periodicity=1 

It works, but does nothing, except making a deep copy of all elements.

Apparently, there is a misunderstanding about what "develop" means: by definition, it applies to periodic lattices (n>1). In the general case, to reproduce a lattice n times, one should use  the '*' operator: `newring = ring * n`. The behaviour is there identical to what you propose: change the `periodicity` attribute if possible, otherwise set it to 1 with a warning.

Otherwise, if you suddenly decide that a lattice is periodic, you may simply set its periodicity to the number you want. And than develop it or not.

To be more clear, I'll suppress the "periods" optional argument. There will be no ambiguity and no need to test for the remainder. The only drawback is to prevent a "partial" development, which is rather unlikely to be needed.